src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1179,30 +1280,32 @@ private sealed class LoaderHolder
                 if (info.Type != typeof(IDataLoader) || info.ArgType != typeof(Options))
                     goto LDone;
 
-                var argsNew = new Options();
-                // Copy the non-core arguments to the new args (we already know that all the core arguments are default).
-                var parsed = CmdParser.ParseArguments(host, CmdParser.GetSettings(host, options, new Options()), argsNew);
-                ch.Assert(parsed);
-                // Copy the core arguments to the new args.
-                if (!CmdParser.ParseArguments(host, loader.GetSettingsString(), argsNew, typeof(ArgumentsCore), msg => ch.Error(msg)))
+                var optionsNew = new Options();
+                // Copy the core arguments to the new options.
+                if (!CmdParser.ParseArguments(host, loader.GetSettingsString(), optionsNew, typeof(Options), msg => ch.Error(msg)))
                     goto LDone;
 
-                cols = argsNew.Columns;
+                // Set the non core arguments in the new options.
```

Comment:
>non core arguments [](start = 27, length = 18)

perhaps we should clarify what are "non core" arguments here ?   would be especially useful for the future.

without #2046 i did not have enough context to know what this meant